[build.webkit.org] Upload WebKit build logs to s3

briannafan opened the pull request:
#### cf619842d9f4be8d6fb994be0a77baa89d06d90b
<pre>
[build.webkit.org] Upload WebKit build logs to s3
<a href="https://bugs.webkit.org/show_bug.cgi?id=270807">https://bugs.webkit.org/show_bug.cgi?id=270807</a>
<a href="https://rdar.apple.com/problem/124396801">rdar://problem/124396801</a>

Reviewed by Aakash Jain.

Filter logs through filter-build-webkit and upload logs to S3.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
(CompileWebKit.start):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success_architecture):

Canonical link: <a href="https://commits.webkit.org/276297@main">https://commits.webkit.org/276297@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78e65d934eaf5d082074387f1d0a4629c8713f55

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44302 "Failed to checkout and rebase branch from PR 25731") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23369 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46737 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/46951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/40328 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46606 "Failed to checkout and rebase branch from PR 25731") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27341 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20766 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/46951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44878 "Failed to checkout and rebase branch from PR 25731") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/20414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/38151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/46951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/39263 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-skia~~](https://ews-build.webkit.org/#/builders/52/builds/2350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/40485 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/39546 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/48559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/19280 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/15832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/48559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/44351 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20642 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/48559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/20970 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6078 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/20267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -413,6 +432,21 @@ def handleExcessiveLogging(self):
+                    extension='txt',
+                    content_type='text/plain',
+                ), UploadFileToS3(
+                    'WebKitBuild/build-log.txt',
```
Better to remove WebKitBuild directory from this path, See: https://commits.webkit.org/274380@main

briannafan replied:
Good catch, done!

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -364,29 +374,38 @@ def start(self):
         self.log_observer = ParseByLineLogObserver(self.parseOutputLine)
         self.addLogObserver('stdio', self.log_observer)
 
+        command = [part.getRenderingFor(self.build) if isinstance(part, WithProperties) else part for part in self.command]
```
Is this needed?

briannafan replied:
We can technically use `getProperty` instead, but this is consistent with how EWS does it.

aj062 replied:
If we can use getProperty, let's use that (and we can change it in EWS as well). That would be more maintainable.

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -347,6 +355,8 @@ def run(self):
 
 class CompileWebKit(shell.Compile, CustomFlagsMixin):
     command = ["perl", "Tools/Scripts/build-webkit", "--no-fatal-warnings", WithProperties("--%(configuration)s")]
```
You can consider renaming `command` to `build_command`, to keep the code between EWS and build.webkit.org similar.

aj062 reviewed (commented):
Also, better to increase the timeout for the CompileWebKit step. Now that there would be less logs, default 20 minutes timeout with logs might not be sufficient. See https://commits.webkit.org/274297@main

You will need to add __init__ method in CompileWebKit for that.

aj062 reviewed (approved):
r+ with a comment (assuming it's tested).

briannafan committed:
[build.webkit.org] Upload WebKit build logs to s3
https://bugs.webkit.org/show_bug.cgi?id=270807
rdar://problem/124396801

Reviewed by Aakash Jain.

Filter logs through filter-build-webkit and upload logs to S3.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
(CompileWebKit.start):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success_architecture):

Canonical link: https://commits.webkit.org/276297@main
